oauthproxy.go

```diff
@@ -810,19 +810,30 @@ func (p *OAuthProxy) getAuthenticatedSession(rw http.ResponseWriter, req *http.R
 // addHeadersForProxying adds the appropriate headers the request / response for proxying
 func (p *OAuthProxy) addHeadersForProxying(rw http.ResponseWriter, req *http.Request, session *sessionsapi.SessionState) {
 	if p.PassBasicAuth {
+		req.Header.Del("Authorization")
+		req.Header.Del("X-Forwarded-Email")
+		req.Header.Del("X-Forwarded-User")
+
 		req.SetBasicAuth(session.User, p.BasicAuthPassword)
 		req.Header["X-Forwarded-User"] = []string{session.User}
 		if session.Email != "" {
 			req.Header["X-Forwarded-Email"] = []string{session.Email}
 		}
```

Comment:
Could setting the `X-Forwarded-Email` header be tidied up with the following?
```
if session.Email != "" {
  req.Header["X-Forwarded-Email"] = []string{session.Email}
} else {
  req.Header.Del("X-Forwarded-Email")
}
```

Reply:
Again same answer as the other one. Yes each section could individually be technically simpler, but it felt more robust to do the same boring thing each time, ie delete all old headers then perform the logic nessary to add new ones.

Reply:
I think from a readability perspective, changing to `if { .. } else { .. } ` actually would make a lot of sense. The logic we want is that we set the header or remove it if we aren't setting it, this is exactly encoded into if/else and would make the code easier to reason with in the future. By doing this we could also get rid of the deletion of headers that are always set and the flow would still be consistent throughout, WDYT @aeijdenberg ?